datafusion/sqllogictest/test_files/string_view.slt

```diff
@@ -1225,6 +1221,48 @@ XiangpengXiangpeng XiangpengXiangpeng XiangpengXiangpeng
 RaphaelRaphael RaphaelRaphael RaphaelRaphael
 NULL NULL NULL
 
+## Ensure no casts for binary operators
+## TODO: https://github.com/apache/datafusion/issues/12180
+# `~` operator (regex match)
+query TT
+EXPLAIN SELECT
```

Comment:
Just parroting some feedback I got on a similar PR: might be nice to include the queries w/o `EXPLAIN` so as to verify correctness not just casts (or in the future no casts 😄).

Reply:
@tshauck Thanks for reviewing. Makes sense to me, I will update it soon.